components/org.wso2.carbon.identity.oauth.endpoint/src/test/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpointTest.java

```diff
@@ -389,6 +389,85 @@ public void testTokenErrorResponse(String errorCode, Object headerObj, int expec
+
+        Map<String, String[]> requestParams = new HashMap<>();
+        requestParams.put(OAuth.OAUTH_GRANT_TYPE, new String[]{GrantType.PASSWORD.toString()});
+        requestParams.put(OAuth.OAUTH_USERNAME, new String[]{USERNAME});
+        requestParams.put(OAuth.OAUTH_PASSWORD, new String[]{"password"});
+
+        mockStatic(LoggerUtils.class);
+        when(LoggerUtils.isDiagnosticLogsEnabled()).thenReturn(true);
+        mockStatic(IdentityTenantUtil.class);
+        when(IdentityTenantUtil.getTenantId(anyString())).thenReturn(-1234);
+        HttpServletRequest request = mockHttpRequest(requestParams, new HashMap<String, Object>());
+        when(request.getHeader(OAuthConstants.HTTP_REQ_HEADER_AUTHZ)).thenReturn(AUTHORIZATION_HEADER);
+        when(request.getHeaderNames()).thenReturn(
+                Collections.enumeration(new ArrayList<String>() {{
+                    add(OAuthConstants.HTTP_REQ_HEADER_AUTHZ);
+                }}));
+
+        spy(EndpointUtil.class);
+        doReturn(REALM).when(EndpointUtil.class, "getRealmInfo");
+        doReturn(oAuth2Service).when(EndpointUtil.class, "getOAuth2Service");
+
+        when(oAuth2Service.issueAccessToken(any(OAuth2AccessTokenReqDTO.class))).thenReturn(oAuth2AccessTokenRespDTO);
+        when(oAuth2AccessTokenRespDTO.getErrorMsg()).thenReturn("Token Response error");
+        when(oAuth2AccessTokenRespDTO.getErrorCode()).thenReturn(errorCode);
+        when(oAuth2AccessTokenRespDTO.getResponseHeaders()).thenReturn(responseHeaders);
+        when(oAuth2AccessTokenRespDTO.getErrorParameterMap()).thenReturn(errorMap);
+
+        mockOAuthServerConfiguration();
+        mockStatic(IdentityDatabaseUtil.class);
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+
+        Map<String, Class<? extends OAuthValidator<HttpServletRequest>>> grantTypeValidators = new Hashtable<>();
+        grantTypeValidators.put(GrantType.PASSWORD.toString(), PasswordValidator.class);
+
+        when(oAuthServerConfiguration.getSupportedGrantTypeValidators()).thenReturn(grantTypeValidators);
+        when(oAuth2Service.getOauthApplicationState(CLIENT_ID_VALUE)).thenReturn("ACTIVE");
+
+        Response response;
+        try {
+            response = oAuth2TokenEndpoint.issueAccessToken(request, new MultivaluedHashMap<String, String>());
+        } catch (InvalidRequestParentException ire) {
+            InvalidRequestExceptionMapper invalidRequestExceptionMapper = new InvalidRequestExceptionMapper();
+            response = invalidRequestExceptionMapper.toResponse(ire);
+        }
+
+        assertNotNull(response, "Token response is null");
+        assertEquals(response.getStatus(), expectedStatus, "Unexpected HTTP response status");
+        assertNotNull(response.getEntity(), "Response entity is null");
+        assertTrue(response.getEntity().toString().contains(expectedErrorCode), "Expected error code not found");
```

Comment:
can we add assert for the header values as well here as we are changing that logic as well

Reply:
Fixed.